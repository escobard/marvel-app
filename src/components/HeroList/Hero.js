import React, { Component } from "react";
import PropTypes from "prop-types";
import { Card, CardHeader, CardText } from "material-ui/Card";
import FlatButton from "material-ui/FlatButton";
import Toggle from "material-ui/Toggle";
import { Link } from "react-router-dom";

import { img } from "../../utils"

export default class Hero extends Component {
  render() {
    let { name, description, urls, thumbnail: { path, extension } } = this.props
      .hero.name
      ? this.props.hero
      : { thumbnail: {} };

    return (
      <Link to={`/hero/${name}`} className="link" params={this.props}>
      <Card className="hero-card">
        <CardHeader
          className="header"
          title={name}
          avatar={img(path, "portrait_medium", extension)}
        />
        <CardText className="description">
          {description
            ? description.substring(0, 40) + "..."
            : "No description available"}
        </CardText>
        
          <FlatButton className="expand" label="Click to expand"/>
        
      </Card>
      </Link>
    );
  }
}

Hero.defaultProps = {
  hero: {}
};

Hero.propTypes = {
  heroes: PropTypes.object
};
