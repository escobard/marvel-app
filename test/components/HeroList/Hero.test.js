import React, { Component } from 'react'
import Hero from '../../../src/components/HeroList/Hero';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('Hero snapshot', () => {
	const initialState = {output:100}
	const mockStore = configureStore()
	let store,container

    beforeEach(()=>{
        store = mockStore(initialState)
        container = shallow(<Hero /> )  
    })

    it('+++ render the Hero component', () => {
       expect(toJson(container)).toMatchSnapshot()
    });

});