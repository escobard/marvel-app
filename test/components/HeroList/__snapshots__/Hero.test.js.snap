// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero snapshot +++ render the Hero component 1`] = `
<Link
  className="link"
  params={
    Object {
      "hero": Object {},
    }
  }
  replace={false}
  to="/hero/undefined"
>
  <Card
    className="hero-card"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardHeader
      avatar="undefined/portrait_medium.undefined"
      className="header"
    />
    <CardText
      className="description"
    >
      No description available
    </CardText>
    <FlatButton
      className="expand"
      disabled={false}
      fullWidth={false}
      label="Click to expand"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
    />
  </Card>
</Link>
`;
